src/instructlab/model/train.py

```diff
@@ -1081,21 +1089,31 @@ def _run_phased_training(
             journal.journal.train_1 = phase_model
         journal.commit()
 
-        _training_phase(
-            train_args=train_args.model_copy(deep=True),
-            torch_args=torch_args,
-            data_path=phase1_data,
-            checkpoint_dir=phase1_checkpoints_dir,
-            num_epochs=phase1_num_epochs,
-            samples_per_save=phase1_samples_per_save,
-            effective_batch_size=phased_phase1_effective_batch_size,
-            # model override not necessary because we expect model to come from ctx.params.model_path.
-        )
-
-        phase_model.ended_at_utc = TrainingJournal.now_utc()
+        failed = False
```

Comment:
Seems like this new logic wrapping _training_phase could be shared.  Either in _training_phase or in a new function.

Reply:
Is this necessary? It seems like the duplication is only happening twice, and so the other section may need to change or do something different in the future. I'm concerned about premature abstraction.

Reply:
My vote would be to abstract it now because:
- With the similar code duplicated, it's really easy for one section to migrate from the other over time.  Wouldn't want that to happen accidentially.
- It's easier to figure out what's duplicated now vs as the code evolves over time

Reply:
Thank you for that explanation, that makes sense. I'll make the change.

Reply:
I've updated this, let me know if it aligns with your suggestion or if it needs to be updated.

Reply:
Looks great, thanks!